Cargo.toml

```diff
@@ -1,26 +1,32 @@
 [package]
 name = "deepgram"
-version = "0.0.1"
+version = "0.1.0"
+authors = ["Deepgram <developers@deepgram.com>"]
 edition = "2021"
+description = "Official Rust SDK for Deepgram's automated speech recognition APIs. "
+license = "MIT"
+repository = "https://github.com/deepgram-devs/deepgram-rust-sdk"
+keywords = ["deepgram", "asr", "transcription", "ai"]
+categories = ["api-bindings", "multimedia::audio"]
 
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
 # TODO Investigate which of these dependencies can go behind features.
 bytes = "1.1.0"
 futures = "0.3.21"
-http = "0.2.6"
+http = "0.2.7"
 pin-project = "1.0.10"
 reqwest = { version = "0.11.10", features = ["json"] }
-serde = { version = "1.0.136", features = [ "derive" ] }
-serde_json = "1.0.79"
-thiserror = "1.0.30"
-tokio = { version = "1.17.0", features = [ "full" ] }
+serde = { version = "1.0.137", features = [ "derive" ] }
+serde_json = "1.0.81"
+thiserror = "1.0.31"
+tokio = { version = "1.19.2", features = [ "full" ] }
 tokio-tungstenite = { version = "0.17.1", features = [ "native-tls" ] }
-tokio-util = { version = "0.7.1", features = [ "codec", "io" ] }
+tokio-util = { version = "0.7.3", features = [ "codec", "io" ] }
 tungstenite = "0.17.2"
 url = "2.2.2"
-uuid = { version = "1.0.0", features = ["serde"] }
+uuid = { version = "1.1.1", features = ["serde"] }
```

Comment:
How strict do we want to  be about versions here?  It looks like we're specifying the current latest for everything.  This could make it harder to integrate the sdk with existing projects.

Reply:
> It looks like we're specifying the current latest for everything.

That's what I did because I wasn't aware of the downsides, and wanted to make sure we're using the version with all the latest fixes and fixes.

Are there any packages that you think we should intentionally specify an older version of (or at least a way for me to figure out which)?

Reply:
Also, if this can't be figured out today, do you think it should hold back the release of version 0.1.0?

Reply:
In general, as I understand it, applications should specify the maximum version they can to make sure they have the most recent versions possible, but libraries should specify the minimum version they require so that more applications can make use of them. This is related to why we don't commit the `Cargo.lock` file for libraries, because our users may ultimately use different versions

Reply:
> so that more applications can make use of them

I've been thinking about this for a while, and I don't see how it applies. SemVer compatibility is supposed to be an [equivalence relation](https://en.wikipedia.org/wiki/Equivalence_relation).

Let's say H is the most recent version of some package P, and L is the lowest version that's compatible to H (example: H = 2.4.3, L = 2.0.0). If a binary crate (that depends on our crate) specifies a version of that package P, that version is compatible with L if and only if it's also compatible with H.

Now, not everyone is perfect with SemVer. Sometimes a "minor" update is published that accidentally includes a breaking change. But it seems like trying to specify a minimum version is also problematic.

When we compile and test our library, Cargo tries to use the most recent SemVer compatible version of every dependency. This means that although we might be promising to be compatible with `tokio` version 1.17.0, we actually might be using features introduced in 1.19.0 without realizing it, and the compiler isn't catching this for us. We could use nightly Rust to try and compile with `minimal-versions`, but that doesn't catch any problems we might have with version 1.18.0.

Specifying the versions the library has been tested against right before a publish seems like the best bet to me. But I'm also not an expert at dependency resolution, and might be weighing the factors wrong.

Reply:
> We could use nightly Rust to try and compile with minimal-versions, but that doesn't catch any problems we might have with version 1.18.0.
> 
> Specifying the versions the library has been tested against right before a publish seems like the best bet to me. But I'm also not an expert at dependency resolution, and might be weighing the factors wrong.

I generally agree with this but I think if we don't attempt to trust the ecosystem we'll make tons of work for ourselves. I'd say to try to compile with minimum versions and deploy with that. If someone reports a bug with a particular version we can update the dependency to be the version above that one (depending on the root cause of their reported issue). That being said I'm probably just being lazy.

Reply:
It's possible for crates to specify a maximum dependency they allow (for example, to maintain a MSRV) so if a binary crate depends on us and another crate, and that crate has specified a maximum version of one of our dependencies that's not compatible, then that user is out of luck and has to pick between our crate and the other, even if we aren't using any of the features from the latest version of the dependency we specified, and could have built fine on the earlier version.

There's a procedure outlined in _Rust for Rustaceans_ for figuring out minimum dependency versions, which involves using `minimal-versions` and setting all our dependencies to their last major version, then incrementing minor versions until you get a successful build, then setting our dependency versions in `Cargo.toml` to whatever got picked in the `Cargo.lock`. Not saying we have to do that right now I think it's overkill for a first release, just relaying it as a general process we'll need to do at some point.

Reply:
+1 for Cargo.toml specifying the minimum version that has the needed API.

Reply:
Okay, I've pushed all the dependencies back to the oldest version that's still SemVer compatible with the latest version and works with our package. I also made the CI workflow run a build with `-Z minimal-versions` so we can always be sure it's working.

> I think it's overkill for a first release

Too bad. Did it anyways.

Reply:
> There's a procedure outlined in _Rust for Rustaceans_ for figuring out minimum dependency versions

It's in Chapter 5 on page 82.